beacon-chain/execution/engine_client.go

```diff
@@ -1013,6 +1080,79 @@ func toBlockNumArg(number *big.Int) string {
 	return hexutil.EncodeBig(number)
 }
 
+// retryReconstructDataColumnSidecars performs periodic retry attempts for getBlobsV2.
+func (s *Service) retryReconstructDataColumnSidecars(ctx context.Context, signedROBlock interfaces.ReadOnlySignedBeaconBlock, blockRoot [fieldparams.RootLength]byte) {
+	retryCtx, cancel := context.WithTimeout(ctx, time.Duration(params.BeaconConfig().SecondsPerSlot)*time.Second) // 1 slot timeout
+	defer cancel()
+
+	// Track active retry
+	s.activeRetries.Store(blockRoot, cancel)
+	defer s.activeRetries.Delete(blockRoot)
+
+	// Ensure we have a valid retry interval
+	retryInterval := s.getBlobsRetryInterval
+	if retryInterval <= 0 {
+		retryInterval = defaultGetBlobsRetryInterval // default fallback
+	}
+	ticker := time.NewTicker(retryInterval)
+	defer ticker.Stop()
+
+	attemptCount := 0
+	startTime := time.Now()
+	log := log.WithField("blockRoot", fmt.Sprintf("%#x", blockRoot))
+
+	for {
+		select {
+		case <-ticker.C:
+			attemptCount++
+
+			// Check if data is now available from any source
+			if s.isDataAlreadyAvailable(retryCtx, blockRoot, signedROBlock) {
```

Comment:
if IsDataAvailable errors this will be false and it will try to reconstruct with data that may be available but was erroring for some reason